cmd/grpcurl/grpcurl.go

```diff
@@ -314,7 +318,7 @@ func main() {
 	if *key != "" && !usetls {
 		fail(nil, "The -key argument can only be used with TLS.")
 	}
-	if (*key == "") != (*cert == "") {
+	if (*key == "") != (*cert == "") && !strings.EqualFold(*certType, "p12") {
 		fail(nil, "The -cert and -key arguments must be used together and both be present.")
 	}
```

Comment:
This is insufficient. With "p12", it still doesn't make sense to provide a `-key` argument but not `-cert`, since the actual kay+cert is loaded from the `-cert` argument. Also, if they _do_ specify both as different files, it should probably be an error with "p12", since the key argument is erroneous (and the current implementation would ignore it).

Also, nowhere is validating the the `-cert-type` flag be "p12" or "pem" as indicated in the help for the flag. So this allows users to set a garbage type, in which case the meaning is unclear. It also misleadingly lets them enter "der" or "eng", but then expects the file to actually be "pem".